### DISCUSSION TITLE
Look at the much more feature rich Litestar as potential alternative to FastAPI

### DISCUSSION THREAD

**Discussion by zro00** [PrefectHQ/prefect#10539] [Ideas]
github link: https://github.com/litestar-org/litestar

Litestar is a new webserver project that aims to have more features out-of-door then what FastAPI provides.

It already provides:
1. Stoplight elements (huge feature improvement when viewing APIs) 
2. Baked in protection --> CFSR (https://docs.litestar.dev/latest/usage/middleware/builtin-middleware.html)
3. Integration with all data validation library (mgspec, attrs and pydantic) 
4. Adding new features at fast clip, very active team and community
5. V2 is even better (just like Prefect the folks at Litestar went a did a redesign) -> this is project that should gain a lot more love than it has
6. Much more modern web features --> htmx
7. More built in features --> native integration for jinja/mako, etc ...

To be perfectly honest FastAPI has stagnated a lot recently. 
Just look at this pull request (active since 2022 for stop light elements and no comment on when it can integrated):
https://github.com/tiangolo/fastapi/pull/5168

**Comment by hasansezertasan** [comment]
> Litestar is a new webserver project that aims to have more features out-of-door then what FastAPI provides.

I don't think this is true. Like Flask, Litestar also aims to move some of its contrib modules. Check this [View 1 · Resolve `contrib` submodule](https://github.com/orgs/litestar-org/projects/12) out.

I don't think HTMX all alone gives a "much more modern web feature" to a "backend framework" but it's good to have.
